From 146ff8c1211da8388192740a1d4c6aa9a0467d9b Mon Sep 17 00:00:00 2001
From: Qiuxu Zhuo <qiuxu.zhuo@intel.com>
Date: Wed, 31 May 2023 10:48:15 +0800
Subject: [PATCH 135/360] EDAC/igen6: Add several Intel Raptor Lake-P SoCs
 support

Mainline: commit d23627a7688f("EDAC/igen6: Add Intel Raptor Lake-P SoCs support")
from: v6.8-rc1

Add several Intel Raptor Lake-P SoC compute die IDs for EDAC support.
These Raptor Lake-P SoCs use similar memory controller and IBECC as
Alder Lake-P SoC but extend the most significant bit of error address
logged in IBECC from bit 38 to bit 45.

Intel-SIG: Upstream commit d23627a7688f ("EDAC/igen6: Add Intel Raptor Lake-P SoCs support").
upstream link: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d23627a7688f&dt=2

Signed-off-by: Qiuxu Zhuo <qiuxu.zhuo@intel.com>
Signed-off-by: Tony Luck <tony.luck@intel.com>
[ Qingdong Shi: amend commit log ]
Signed-off-by: Qingdong Shi <qingdong.shi@intel.com>
---
 drivers/edac/igen6_edac.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/edac/igen6_edac.c b/drivers/edac/igen6_edac.c
index c25790a9da2d..286e9603756c 100644
--- a/drivers/edac/igen6_edac.c
+++ b/drivers/edac/igen6_edac.c
@@ -294,6 +294,13 @@ static int get_mchbar(struct pci_dev *pdev, u64 *mchbar)
 	return 0;
 }
 
+/* Compute die IDs for Raptor Lake-P with IBECC */
+#define DID_RPL_P_SKU1	0xa706
+#define DID_RPL_P_SKU2	0xa707
+#define DID_RPL_P_SKU3	0xa708
+#define DID_RPL_P_SKU4	0xa716
+#define DID_RPL_P_SKU5	0xa718
+
 static bool ehl_ibecc_available(struct pci_dev *pdev)
 {
 	u32 v;
-- 
2.20.1

